src/components/TimePicker24Hour/TimePicker24Hour.jsx

```diff
@@ -39,10 +51,18 @@ const TimePicker24Hour = (props) => {
     const displayTime = e.target.value;
     const newTime = moment(displayTime, "HH:mm").format("HH:mm");
     if (isValidTime(newTime)) {
-      setWarning(false);
+      if (customValidation) {
+        customValidation(newTime);
+      } else {
+        setWarning(false);
+      }
       setTime(newTime);
     } else {
       setWarning(true);
+      setWarningText(
+        invalidText || "Please enter a valid time in 24-hr format"
+      );
+      actionForInvalidTime(true);
```

Comment:
undefined check for actionForInvalidTime callback prop is missed here

Reply:
added undefined check